### PR TITLE
[playlist] Add support for reading XSPF Playlists

### PR DESCRIPTION
**Pull request by tylerszabo**
## Description

Allows for .xspf files to be read as playlists

## Motivation and Context

I wanted to get Kodi to read playlists I had made using VLC. I searched to see if the functionality existed and I came upon this thread https://forum.kodi.tv/showthread.php?tid=209910. While I can convert, I thought it'd be handy to have Kodi also support this (powerful and extensible) format.

## How Has This Been Tested?

Unit tests were created and run on Windows 10 & Linux.
Manual testing was done one Windows 10 64-bit and Ubuntu 18.04.1 64-bit VM.
1. Create a music directory source that contains audio files in subdirectories and playlists that specify paths to those files based on a Windows system's file structure and some from a Linux system's file structure.
2. Add a music directory as a music source
3. Open the file "Files" browser for the music source and locate the playlist files
4. Open the various playlist files and observe how they are populated when the file exists in the given filesystem and are playable. (Note that a playlist that includes location paths but no file exists at that location (on that system) are still loaded, but the respective entries are skipped, this is the behavior specified XSPF.)

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [x] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/playlists/PlayListXSPF.cpp [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+	</trackList>
+</playlist>
+*/
+
+#define FILE_PROTOCOL_WITH_ROOT "file:///"
+#define LOCATION_TAGNAME "location"
+#define PLAYLIST_TAGNAME "playlist"
+#define TITLE_TAGNAME "title"
+#define TRACK_TAGNAME "track"
+#define TRACKLIST_TAGNAME "trackList"
```
Comment (Rechi) [inline xbmc/playlists/PlayListXSPF.cpp]: make those `static std::string` or `static constexpr char*`

File: xbmc/playlists/PlayListXSPF.cpp [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  CXBMCTinyXML xmlDoc;
+  const char * errorStr = "";
+  int errorRow = 0;
+  int errorCol = 0;
+
+  const char * label = "";
+  const char * location = "";
+  int offset = 0;
+  std::string localpath;
+  CURL uri;
```
Comment (Rechi) [inline xbmc/playlists/PlayListXSPF.cpp]: This isn't C89 code, so declare the variables where they are actually used.
Also use `std::string` instead of `char*`

File: xbmc/playlists/PlayListXSPF.cpp [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+      TiXmlElement* pCurTitle = pCurTrack->FirstChildElement(TITLE_TAGNAME);
+      if (pCurTitle)
+      {
+        label = pCurTitle->GetText();
+      }
+
+      CFileItemPtr newItem(new CFileItem(label));
+
+      location = pCurLocation->GetText();
+      CLog::Log(LOGINFO, "Interpreting location=%s", location);
```
Comment (Rechi) [inline xbmc/playlists/PlayListXSPF.cpp]: this spams the log

File: xbmc/playlists/PlayListXSPF.cpp [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+      if (StringUtils::StartsWith(location, FILE_PROTOCOL_WITH_ROOT))
+      {
+#ifdef TARGET_WINDOWS
+        // omit leading / from "file:///C:/Users/..."
+	      offset = sizeof(FILE_PROTOCOL_WITH_ROOT) - 1;
+#else
+        // include leading / from "file:///home/..."
+	      offset = sizeof(FILE_PROTOCOL_WITH_ROOT) - 2;
+#endif
```
Comment (Rechi) [inline xbmc/playlists/PlayListXSPF.cpp]: this looks hacky
indentation is wrong

File: xbmc/playlists/PlayListXSPF.cpp [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+      }
+
+      if (!localpath.empty())
+      {
+#ifdef TARGET_WINDOWS
+        StringUtils::Replace(localpath, "/", "\\");
+        localpath = URIUtils::CanonicalizePath(localpath, '\\');
+#else
+        localpath = URIUtils::CanonicalizePath(localpath, '/');
+#endif
```
Comment (Rechi) [inline xbmc/playlists/PlayListXSPF.cpp]: this looks hacky

File: xbmc/playlists/PlayListXSPF.cpp [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+        newItem->SetPath(localpath);
+        CLog::Log(LOGINFO, "Adding item to playlist label=%s, localpath=%s", label, localpath);
```
Comment (Rechi) [inline xbmc/playlists/PlayListXSPF.cpp]: spams the log

File: xbmc/playlists/PlayListXSPF.cpp [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+      }
+      else
+      {
+        newItem->SetURL(uri);
+        CLog::Log(LOGINFO, "Adding item to playlist label=%s, uri=%s", label, uri.Get());
```
Comment (Rechi) [inline xbmc/playlists/PlayListXSPF.cpp]: spams the log

File: xbmc/playlists/PlayListXSPF.cpp [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+  int offset = 0;
+  std::string localpath;
+  CURL uri;
+
+  TiXmlElement* pPlaylist = nullptr;
+  TiXmlElement* pTracklist = nullptr;
+  TiXmlElement* pPlaylistTitle = nullptr;
+  TiXmlElement* pCurTrack = nullptr;
+
+  CLog::Log(LOGINFO, "ZZ Playlist=%s", strFileName);
```
Comment (Rechi) [inline xbmc/playlists/PlayListXSPF.cpp]: spams the log

File: xbmc/playlists/PlayListXSPF.cpp (hunk 4 above)
Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: Dang, that indentation slipped by. As for the pattern what's going on is the "file://" [URI scheme](https://en.wikipedia.org/wiki/File_URI_scheme) has no host and goes straight into an absolute path. On Windows the path needs to start at the drive and on Linux the path needs to start with the leading "/". Since the "/" is already in the string I opted to grab it. What do you think of this, instead:

```c++
#ifdef TARGET_WINDOWS
	localpath = "";
#else
        localpath = "/";
#endif
        localpath += CURL::Decode(location + sizeof(FILE_PROTOCOL_WITH_ROOT) - 1);
```

Comment (Rechi) [reply xbmc/playlists/PlayListXSPF.cpp]: looks better
please avoid pointer arithmetic with and also the sizeof, better use length and substr method from std::string

File: xbmc/playlists/PlayListXSPF.cpp (hunk 2 above)
Comment (peak3d) [reply xbmc/playlists/PlayListXSPF.cpp]: I use sometimes goto, too. They can help improoving readability

File: xbmc/playlists/PlayListXSPF.cpp [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+			<title>KEXP.ORG 90.3FM - where the music matters</title>
+      <info>https://kexp.org/</info>
+      <image>https://www.kexp.org/static/assets/img/logo-white.svg</image>
+		</track>
+    <track><location>Relative/Path/To.mp3</location></track>
+    <track><location>file:///music/song_1.ogg</location></track>
+    <track><location>file:///music/song_2.flac</location></track>
+    <track><location>file:///music/song_3.mp3</location></track>
+	</trackList>
+</playlist>
```
Comment (ace20022) [inline xbmc/playlists/PlayListXSPF.cpp]: The indentation in the above block looks odd.

File: xbmc/playlists/PlayListXSPF.cpp (hunk 5 above)
Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: I'll take another pass at this. Is there a preferred format for storing paths in a platform-independent way? I found the `file://` protocol wasn't working as expected (was hoping that CFileItem->SetPath() on a `file://` would resolve correctly). Otherwise I'll see if I can keep to forward slashes with `URIUtils::CanonicalizePath(localpath, '/');`.

File: xbmc/playlists/PlayListXSPF.cpp (hunk 2 above)
Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: Good point on goto being a bit mixed. I'll just refactor to remove them - even if they look clean in this case by having them may encourage attempting to copy the pattern without good reason (and the compiler errors from the uninitialized variables might be painful for someone that's not used to the pattern). It might be handy to add to the style guide to "avoid goto" then I'd just avoid it even if I see the other places.

File: xbmc/playlists/PlayListXSPF.cpp (hunk 4 above)
Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: Ah, yes, that is cleaner and more obvious, thanks :)

File: xbmc/playlists/PlayListXSPF.cpp (hunk 9 above)
Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: Oh shoot, it's the "tabs" I thought I replaced them with spaces. Thanks for catching this; I'll do a grep-pass to see if I did this in any other files.

File: xbmc/playlists/PlayListXSPF.cpp (hunk 4 above)
Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: I removed the `sizeof` math off a pointer in lieu of substr with a number and a comment. (I tried to use URL.cpp parsing as a style-guide).

File: xbmc/playlists/PlayListXSPF.cpp (hunk 5 above)
Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: It didn't get a lot better, but I'm hoping this is a little more clear. I've added comments describing why this pattern is being used.

File: xbmc/playlists/PlayListXSPF.cpp [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+  if (!pPlaylist)
+  {
+    CLog::Log(LOGERROR, "Error parsing XML file %s: missing root element %s", strFileName.c_str(), PLAYLIST_TAGNAME);
+    return false;
+  }
+
+  TiXmlElement* pTracklist = pPlaylist->FirstChildElement(TRACKLIST_TAGNAME);
+  if (!pTracklist)
+  {
+    CLog::Log(LOGERROR, "Error parsing XML file %s (%d, %d): missing element %s", strFileName.c_str(), pPlaylist->Row(), pPlaylist->Column(), TRACKLIST_TAGNAME);
```
Comment (Rechi) [inline xbmc/playlists/PlayListXSPF.cpp]: For me a missing element isn't related to any line or column. Does this even give useful locations?

File: xbmc/playlists/PlayListXSPF.cpp [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+    m_strPlayListName = pPlaylistTitle->GetText();
+  }
+
+  TiXmlElement* pCurTrack = pTracklist->FirstChildElement(TRACK_TAGNAME);
+
+  while (pCurTrack)
+  {
+    std::string label = "";
+    std::string location = "";
+    std::string localpath = "";
```
Comment (Rechi) [inline xbmc/playlists/PlayListXSPF.cpp]: `= ""` isn't necessary for std::string, that's what the default constructor does anyway.

File: xbmc/playlists/PlayListXSPF.cpp (hunk 10 above)
Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: I felt that it would help locate the "playlist" tag that it was looking inside (in the case where you potentially have something malformed like `<playlist/><playlist>...<playlist>` and it fails to load. Since the information was easy to get at I figured why not put it in the log. If you think it's less clean it's just as easy to remove. I have no preference.

Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: Good point on how useful it is, though. The more I think about it the more I think unless I'm sure it's always reliable (and I'm not sure of that) it's more likely to be misleading.

File: xbmc/playlists/PlayListXSPF.cpp [hunk 12]
```diff
@@ -0,0 +1,150 @@
+/*
+ *  Copyright (C) 2018 Tyler Szabo
```
Comment (garbear) [inline xbmc/playlists/PlayListXSPF.cpp]: Could you change this to:

```
/*
 *  Copyright (C) 2018 Tyler Szabo
 *  Copyright (C) 2018 Team Kodi
 *  This file is part of Kodi - https://kodi.tv
 *
 *  SPDX-License-Identifier: GPL-2.0-or-later
 *  See LICENSES/README.md for more information.
 */
```

File: xbmc/playlists/PlayListXSPF.cpp [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+    </track>
+    <track><location>Relative/Path/To.mp3</location></track>
+    <track><location>file:///music/song_1.ogg</location></track>
+    <track><location>file:///music/song_2.flac</location></track>
+    <track><location>file:///music/song_3.mp3</location></track>
+  </trackList>
+</playlist>
+*/
+
+static constexpr char const * LOCATION_TAGNAME = "location";
```
Comment (garbear) [inline xbmc/playlists/PlayListXSPF.cpp]: This diverges from our coding style... The pointer modifies the following ID, so I'd prefer `char const *LOCATION_TAGNAME`

File: xbmc/playlists/PlayListXSPF.cpp [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+static constexpr char const * TITLE_TAGNAME = "title";
+static constexpr char const * TRACK_TAGNAME = "track";
+static constexpr char const * TRACKLIST_TAGNAME = "trackList";
+
+
+CPlayListXSPF::CPlayListXSPF(void) = default;
+
+CPlayListXSPF::~CPlayListXSPF(void) = default;
+
+
```
Comment (garbear) [inline xbmc/playlists/PlayListXSPF.cpp]: Single line between functions

File: xbmc/playlists/PlayListXSPF.cpp [hunk 15]
```diff
@@ -0,0 +1,150 @@
...
+  }
+
+  TiXmlElement* pCurTrack = pTracklist->FirstChildElement(TRACK_TAGNAME);
+
+  while (pCurTrack)
+  {
+    std::string label;
+    std::string location;
+    std::string localpath;
+    CURL uri;
```
Comment (garbear) [inline xbmc/playlists/PlayListXSPF.cpp]: Can you move these to above their first use?

File: xbmc/playlists/PlayListXSPF.cpp [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+
+    TiXmlElement* pCurLocation = pCurTrack->FirstChildElement(LOCATION_TAGNAME);
+    if (pCurLocation)
+    {
+
+      TiXmlElement* pCurTitle = pCurTrack->FirstChildElement(TITLE_TAGNAME);
+      if (pCurTitle)
+      {
+        label = pCurTitle->GetText();
+      }
```
Comment (garbear) [inline xbmc/playlists/PlayListXSPF.cpp]: brackets {} not required for one-line if or if/else statements

File: xbmc/playlists/PlayListXSPF.cpp [hunk 17]
```diff
@@ -0,0 +1,150 @@
...
+#include "utils/log.h"
+#include "utils/URIUtils.h"
+#include "utils/XBMCTinyXML.h"
+#include "URL.h"
+
+using namespace PLAYLIST;
+
+/*
+
+Specification: http://www.xspf.org/specs/
```
Comment (garbear) [inline xbmc/playlists/PlayListXSPF.cpp]: Do we want this in our codedocs? If so, then it should be in doxygen format in the header.

File: xbmc/playlists/test/TestPlayListXSPF.cpp [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+#include "test/TestUtils.h"
+#include "utils/URIUtils.h"
+#include "FileItem.h"
+#include "URL.h"
+
+#include "gtest/gtest.h"
+
+using namespace PLAYLIST;
+
+
```
Comment (garbear) [inline xbmc/playlists/test/TestPlayListXSPF.cpp]: some more double-newlines in this file

**Comment by garbear** [comment]
Looking better

File: xbmc/playlists/PlayListXSPF.cpp (hunk 17 above)
Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: I have a mixed opinion on it. I don't know the header is the right place because it doesn't provide much information to the caller (the caller doesn't need to know how to parse this) it's helpful to future editors to know what the document is expected to look like (see the other Playlist cpp files). However, because there's the test case we have an example (that, IMO is in a better location). I'll remove it but it might be worth adding back in for consistency among other playlists.

File: xbmc/playlists/PlayListXSPF.cpp (hunk 13 above)
Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: Thanks for clarifying. May I put this in `/docs/CODE_GUIDELINES.md`?

Comment (pkerling) [reply xbmc/playlists/PlayListXSPF.cpp]: There are a few more things missing in that document - I'll take it from here

File: xbmc/playlists/PlayListXSPF.cpp [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+static constexpr char const* PLAYLIST_TAGNAME = "playlist";
+static constexpr char const* TITLE_TAGNAME = "title";
+static constexpr char const* TRACK_TAGNAME = "track";
+static constexpr char const* TRACKLIST_TAGNAME = "trackList";
+
+CPlayListXSPF::CPlayListXSPF(void) = default;
+
+CPlayListXSPF::~CPlayListXSPF(void) = default;
+
+static inline const std::string GetXMLText(const TiXmlElement* const pXmlElement)
```
Comment (pkerling) [inline xbmc/playlists/PlayListXSPF.cpp]: Sidenote: You can remove the `inline`, it doesn't actually do anything here

Also why is the return value `const`?

Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: Good point! Fixed, thanks :)

I initially wrote this as `const std::string&` out of habit (got rid of the reference because of a [C++11 nuance](https://stackoverflow.com/questions/10231349/are-the-days-of-passing-const-stdstring-as-a-parameter-over)).

Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: Good catch. I initially was returning `const std::string&` due to a auto-pilot thinking of it as an accessor except that made no sense because it existed inside the function's stack.

File: xbmc/utils/Mime.cpp [hunk 20]
```diff
@@ -486,10 +486,11 @@ static std::map<std::string, std::string> fillMimeTypes()
   mimeTypes.insert(std::pair<std::string, std::string>("xlw",       "application/excel"));
   mimeTypes.insert(std::pair<std::string, std::string>("xm",        "audio/xm"));
   mimeTypes.insert(std::pair<std::string, std::string>("xml",       "text/xml"));
+  mimeTypes.insert(std::pair<std::string, std::string>("x-png",     "image/png"));
   mimeTypes.insert(std::pair<std::string, std::string>("xmz",       "xgl/movie"));
   mimeTypes.insert(std::pair<std::string, std::string>("xpix",      "application/x-vnd.ls-xpix"));
   mimeTypes.insert(std::pair<std::string, std::string>("xpm",       "image/xpm"));
-  mimeTypes.insert(std::pair<std::string, std::string>("x-png",     "image/png"));
```
Comment (garbear) [inline xbmc/utils/Mime.cpp]: Why is this line moved?

File: xbmc/playlists/PlayListXSPF.cpp [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+static constexpr char const* PLAYLIST_TAGNAME = "playlist";
+static constexpr char const* TITLE_TAGNAME = "title";
+static constexpr char const* TRACK_TAGNAME = "track";
+static constexpr char const* TRACKLIST_TAGNAME = "trackList";
+
+CPlayListXSPF::CPlayListXSPF(void) = default;
+
+CPlayListXSPF::~CPlayListXSPF(void) = default;
+
+static std::string GetXMLText(const TiXmlElement* const pXmlElement)
```
[the remark this answers is not in the document]
Comment (pkerling) [reply xbmc/playlists/PlayListXSPF.cpp]: > Also, wrap this help in an anonymous namepace to avoid symbol collisions

Marking a global function `static` is equivalent to an anonymous namespace

File: xbmc/utils/Mime.cpp (hunk 20 above)
Comment (tylerszabo) [reply xbmc/utils/Mime.cpp]: Artifact of sort, it actually should have moved higher but I missed I hadn't selected all of the 'x' range. Looking at it again, though, it does make sense to be at `xp` despite `x-p` lexically sorting before `xi`.

Comment (tylerszabo) [reply xbmc/utils/Mime.cpp]: Artifact of sort; it actually should have moved higher but I missed I hadn't selected all of the `x` range. Looking at it again, though, it does make sense to be at `xp` despite `x-p` lexically sorting before `xi`.

File: xbmc/playlists/PlayListXSPF.cpp (hunk 21 above)
[the remark this answers is not in the document]
Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: This discussion collapsed with my latest revision; but I'd still like to know which pattern to use. It seems according to [this SO post](https://stackoverflow.com/questions/154469/unnamed-anonymous-namespaces-vs-static-functions) that the anonymous namespace might be a good choice; though I'd argue the best choice is the one that'll be most consistent with the direction the codebase is intending to go.

[the remark this answers is not in the document]
Comment (pkerling) [reply xbmc/playlists/PlayListXSPF.cpp]: We do not have guidelines on this (yet), so you are free to do what you like - as it is a new file you don't have to match up with anything. My personal choice would be anonymous namespaces though, and from the looks of it @garbear seems to agree.

[the remark this answers is not in the document]
Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: I like the anonymous namespace solution; it also seems like it'd be good to put the constants in there, too.

File: xbmc/playlists/PlayListXSPF.cpp (hunk 12 above)
Comment (garbear) [reply xbmc/playlists/PlayListXSPF.cpp]: you didn't follow my instructions ;)

Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: I had to look at this 3 times to realize I missed the `This file is part of Kodi - https://kodi.tv` line. Sorry about that.

Comment (tylerszabo) [reply xbmc/playlists/PlayListXSPF.cpp]: I verified the lines are verbatim in 3 of the 4 files I added. `PlayListXSPF.h` has Team Kodi only.

**Comment by tylerszabo** [comment]
I'd like to thank everyone for reviewing this change - I realized it took time to ensure that the change would perform correctly and the style conformed to the way project's goals. I think this project can credit its readability to the care you take in keeping code quality high.

File: xbmc/playlists/PlayListXSPF.cpp [hunk 22]
```diff
@@ -0,0 +1,127 @@
...
+#include "utils/URIUtils.h"
+#include "utils/XBMCTinyXML.h"
+#include "URL.h"
+
+using namespace PLAYLIST;
+
+namespace
+{
+
+static constexpr char const* LOCATION_TAGNAME = "location";
```
Comment (pkerling) [inline xbmc/playlists/PlayListXSPF.cpp]: Remove `static` - you already put it in an anonymous namespace, which has the same effect

File: xbmc/playlists/PlayListXSPF.cpp [hunk 23]
```diff
@@ -0,0 +1,127 @@
...
+static constexpr char const* PLAYLIST_TAGNAME = "playlist";
+static constexpr char const* TITLE_TAGNAME = "title";
+static constexpr char const* TRACK_TAGNAME = "track";
+static constexpr char const* TRACKLIST_TAGNAME = "trackList";
+
+static std::string GetXMLText(const TiXmlElement* pXmlElement)
```
Comment (pkerling) [inline xbmc/playlists/PlayListXSPF.cpp]: same

File: xbmc/platform/darwin/ios/Info.plist.in [hunk 24]
```diff
@@ -320,6 +320,7 @@
           <string>aif</string>
           <string>aiff</string>
           <string>wpl</string>
+          <string>xspf</string>
```
Comment (MartijnKaijser) [inline xbmc/platform/darwin/ios/Info.plist.in]: Might also need to be added to UWP (and other platforms) acceptable extensions we can open

Comment (pkerling) [reply xbmc/platform/darwin/ios/Info.plist.in]: @rechi @Paxxi maybe?

Linux doesn't need it afaik

Comment (MartijnKaijser) [reply xbmc/platform/darwin/ios/Info.plist.in]: https://github.com/xbmc/xbmc/blob/master/tools/windows/packaging/uwp/package.appxmanifest.in

**Review by notspiff — APPROVED** [review]
Looks great.

**Comment by MartijnKaijser** [comment]
Thx for the patience
